static/app/views/automations/components/editAutomationActions.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import {AutomationFixture} from 'sentry-fixture/automations';
+import {OrganizationFixture} from 'sentry-fixture/organization';
+
+import {
+  render,
+  renderGlobalModal,
+  screen,
+  userEvent,
+  within,
+} from 'sentry-test/reactTestingLibrary';
+
+import {EditAutomationActions} from './editAutomationActions';
+
+describe('EditAutomationActions', () => {
+  it('calls delete mutation when deletion is confirmed', async () => {
+    const automation = AutomationFixture();
+    const organization = OrganizationFixture();
+
+    const mockDeleteAutomation = MockApiClient.addMockResponse({
+      url: `/organizations/${organization.slug}/workflows/${automation.id}/`,
+      method: 'DELETE',
+    });
+
+    const {router} = render(<EditAutomationActions automation={automation} />);
+    renderGlobalModal();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Delete'}));
+
+    // Confirm the deletion
+    const dialog = await screen.findByRole('dialog');
+    await userEvent.click(within(dialog).getByRole('button', {name: 'Delete'}));
+
+    expect(mockDeleteAutomation).toHaveBeenCalledWith(
+      `/organizations/${organization.slug}/workflows/${automation.id}/`,
+      expect.anything()
+    );
+
+    // Redirect to the monitors list
+    expect(router.location.pathname).toBe(
```

Comment:
Since this happens after an async action, this might be less prone to flaking in the future if you do `await waitFor() => {...}`